### DISCUSSION TITLE
How to see package upgrade history?

### DISCUSSION THREAD

**Discussion by abitrolly** [Homebrew/discussions#5807] [Everyday usage] [answered]
Is there a way to see when a package was upgraded and from which version to which?

**Comment by carlocab** [comment]
You can try looking at the Git log.

**Reply by abitrolly** [reply]
Do you mean this log?
```sh
git -C /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core log
```
That doesn't show which version I had.

**Reply by gromgit** [reply, marked as answer]
If you're looking for an install log like apt's `history.log`, then as far as I know, Homebrew doesn't write anything like that, and even `history.log` doesn't tell you what old package versions are overwritten.

This is something you can probably cobble together yourself by parsing the current output of `brew upgrade` though.
